server/testutil/app/app.go

```diff
@@ -0,0 +1,220 @@
+package app
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	bazelgo "github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+const (
+	// readyCheckPollInterval determines how often to poll BuildBuddy server to check
+	// whether it's up and running.
+	readyCheckPollInterval = 500 * time.Millisecond
+	// readyCheckTimeout determines how long to wait until giving up on waiting for
+	// BuildBuddy server to become ready. If this timeout is reached, the test case
+	// running the server will fail with a timeout error.
+	readyCheckTimeout = 30 * time.Second
+)
+
+type App struct {
+	httpPort       int
+	gRPCPort       int
+	monitoringPort int
+	mu             sync.Mutex
```

Comment:
A mistake I have made before (in our caching code) was initializing a sync.Mutex like this in a Cache struct and then creating a new instance of a Cache struct and copying over the mutex (in WithPrefix). Since it wasn't a pointer, I then had two mutexes! Which are not twice as good as one, let me tell you :P

Anyway, I think you're fine here, but something to watch out for.